### PR TITLE
SNOW-533748 partner application name injection

### PR DESCRIPTION
**Pull request by sfc-gh-madkins**
SNOW-533748
This is an enhancement to the python connector to allow snowflake to track partner driven consumption by searching for a partner defined environment variable in circumstances where a partner provides managed infrastructure, but a customer manual downloads the connector via pip.

### CODE REVIEW

File: src/snowflake/connector/connection.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def DefaultConverterClass():
     "authenticator": (DEFAULT_AUTHENTICATOR, (type(None), str)),
     "mfa_callback": (None, (type(None), Callable)),
     "password_callback": (None, (type(None), Callable)),
-    "application": (CLIENT_NAME, (type(None), str)),
+    "application": (os.environ.get('SNOWFLAKE_PARTNER', CLIENT_NAME), (type(None), str)),
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/connection.py:153]: We usually prefix env variables with `SF_`, let's keep to that.
```suggestion
    "application": (os.environ.get('SF_PARTNER', CLIENT_NAME), (type(None), str)),
```

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]
This also needs a test, someone should add one

File: src/snowflake/connector/constants.py [hunk 2]
```diff
@@ -255,3 +255,6 @@ class IterUnit(Enum):
 
 S3_CHUNK_SIZE = 8388608  # boto3 default
 AZURE_CHUNK_SIZE = 4 * megabyte
+
+# TODO: all env variables definitions should be here
+ENV_VAR_PARTNER = "SNOWFLAKE_PARTNER"
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/constants.py:260]: Same as what I said before @sfc-gh-madkins
```suggestion
ENV_VAR_PARTNER = "SF_PARTNER"
```

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

**Comment by sfc-gh-madkins** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by sfc-gh-mkeller — APPROVED** [review]

**Pull request merged**